qutebrowser/keyinput/basekeyparser.py

```diff
@@ -104,33 +191,36 @@ def _match_key(self, sequence):
         """
         assert sequence
         assert not isinstance(sequence, str)
-        result = QKeySequence.NoMatch
-
-        for seq, cmd in self.bindings.items():
-            assert not isinstance(seq, str), seq
-            match = sequence.matches(seq)
-            if match == QKeySequence.ExactMatch:
-                return match, cmd
-            elif match == QKeySequence.PartialMatch:
-                result = QKeySequence.PartialMatch
-
-        return result, None
-
-    def _match_without_modifiers(self, sequence):
+        override_result = (QKeySequence.NoMatch, None)
+        overrides = config.instance.get(
+            'bindings.commands', url=url, fallback=False)
+        if overrides != configutils.UNSET:
+            # TODO add caching here, or simply store binding trie in config
+            # system instead of converting it on the fly.
```

Comment:
Note:

1. It's impossible to store binding trie in config system because it's not representable as YAML.
2. We can make the type of `bindings.commands` = `BindingTrie` instead of `Dict[Key, String]`, but it's still necessary to convert it every time it's get, unless the config cache is used. However, currently the config cache doesn't support per domain configuration (that needs to be fixed)